### ISSUE TITLE
Installing rustup with complete profile prevents adding 'rust-src' component

### ISSUE THREAD

**Issue by Neowizard** [rust-lang/rustup#2134]
**Problem**
I installed rustup (rustup 1.20.2 (13979c968 2019-10-16)) following the instructions in rustup.rs. During installation, I chose the 'Complete Profile'. Afterwards I created a cargo project in Clion (and later again in IntelliJ) with the Rust plugin. The IDE failed to auto complete since it failed to install the 'rust-src' component.
I tried to add 'rust-src' manually with `rustup componenet add rust-src`, but received the error:
```error: component 'rust-src' for target 'x86_64-unknown-linux-gnu' is unavailable for download for channel stable```

Uninstalled rustup with:
```
rustup uninstall stable
rustup self uninstall
```
and reinstalled it using the default profile. Everything works.
**Steps**

1. Install rustup using `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
2. During installation select complete profile (selecting defaults for all other options)
3. Try to add 'rust-src' componenet

**Notes**

Output of `rustup --version`: rustup 1.20.2 (13979c968 2019-10-16)
Output of `rustup show`:
```Default host: x86_64-unknown-linux-gnu
rustup home:  /home/---/.rustup

stable-x86_64-unknown-linux-gnu (default)
rustc 1.39.0 (4560ea788 2019-11-04)
```

OS: Linux Mint 19.1 Tessa

**Comment by kinnison** [comment]
Hi @Neowizard 

I'm sorry you're encountering this -- this is a bug in `rustup` 1.20.x to do with the installation of a target non-specific component (`rust-src`)  The issue is fixed in master and will be in `1.21.0` which is close to being released https://github.com/rust-lang/rustup/milestone/2

In the meantime, if you, in a terminal, run `rustup component remove rust-src` then it should clear out the broken component and let CLion reinstall it for you properly.  For reference, the `complete` profile is mostly meant for developers of Rust itself, not developers who use Rust, the `default` profile is likely more use for you unless you intend to work on clippy, miri, or similar tooling, rather than simply with it.

Good luck, and assuming that fixes things for you, please close the issue.

D.

**Comment by Neowizard** [comment]
Using the 'default' profile works great for me, thanks. 
Closing the issue.

P.S. Happy to hear the issue is already fixed in master. Perhaps the `rustup` install script should mention who the 'complete' profile is targeted towards. I just figured it'll include extra components and maybe save me more installation time in the future.
